### PR TITLE
dataset descriptions2

### LINKED ISSUE #337: datasets need 'description' entry (was datasets need DESCR entry)
In datasets like the yeo atlas, or the craddock atlas, it is very hard to figure out which keys mean what.

These datasets need a 'DESCR' entry, describing that, and maybe the names of the keys could be improved.

### PR DESCRIPTION
**Pull request by banilo**
next piece to fix #337

### CODE REVIEW

File: nilearn/datasets.py [hunk 1]
```diff
@@ -764,7 +764,7 @@ def fetch_craddock_2011_atlas(data_dir=None, url=None, resume=True, verbose=1):
     See http://www.nitrc.org/projects/cluster_roi/ for more information
     on this parcellation.
     """
-
+    module_path = os.path.dirname(__file__)
```
Comment (AlexandreAbraham) [inline nilearn/datasets.py]: This could be a module scope variable.

File: nilearn/datasets.py [hunk 2]
```diff
@@ -787,7 +787,12 @@ def fetch_craddock_2011_atlas(data_dir=None, url=None, resume=True, verbose=1):
     sub_files = _fetch_files(data_dir, filenames, resume=resume,
                              verbose=verbose)
 
-    params = dict(zip(keys, sub_files))
+    with open(os.path.join(module_path, 'description', 'craddock.rst'))\
```
Comment (AlexandreAbraham) [inline nilearn/datasets.py]: A good practice would be to use "<dataset_name>.rst" as name (here craddock_2011) and create a function to get directly the rst content.

Comment (banilo) [inline nilearn/datasets.py]: Year, realized this one!

File: nilearn/datasets.py [hunk 3]
```diff
@@ -787,7 +796,10 @@ def fetch_craddock_2011_atlas(data_dir=None, url=None, resume=True, verbose=1):
     sub_files = _fetch_files(data_dir, filenames, resume=resume,
                              verbose=verbose)
 
-    params = dict(zip(keys, sub_files))
+    fdescr = _get_dataset_descr('craddock')
```
Comment (AlexandreAbraham) [inline nilearn/datasets.py]: Again, please use the unique name given to each dataset.

**Comment by AlexandreAbraham** [comment]
You should fix the error raised by landscape.

**Comment by banilo** [comment]
Ok, just updated this one!

**Comment by lesteve** [comment]
> Ok, just updated this one!

Failing tests, something related with craddock_**2011** vs craddock_**2012**

**Comment by lesteve** [comment]
BTW I disabled landscape-bot commenting on the issues. Shout if you really want it back !

File: nilearn/datasets.py [hunk 4]
```diff
@@ -764,13 +796,12 @@ def fetch_craddock_2011_atlas(data_dir=None, url=None, resume=True, verbose=1):
     See http://www.nitrc.org/projects/cluster_roi/ for more information
     on this parcellation.
     """
-
     if url is None:
         url = "ftp://www.nitrc.org/home/groups/cluster_roi/htdocs" \
               "/Parcellations/craddock_2011_parcellations.tar.gz"
```
Comment (lesteve) [inline nilearn/datasets.py]: @AlexandreAbraham for consistency reasons, should we change 2011 to 2012 in nitrc too? I guess we'd want to keep the old (i.e. 2011) naming for a while for backward compatibility reasons ...

File: nilearn/datasets.py [hunk 5]
```diff
@@ -529,6 +529,38 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
...
+
+    try:
+        with open(os.path.join(module_path, 'description', fname + '.rst'))\
+                as rst_file:
+            descr = rst_file.read()
+    except Exception as e:
+        descr = ''
+
+    if descr == '':
+        print("Warning: Could not find dataset description!")
```
Comment (lesteve) [inline nilearn/datasets.py]: You probably want to use warning.warn rather than a print statement

File: nilearn/datasets.py (hunk 4 above)
Comment (lesteve) [inline nilearn/datasets.py]: Hmmm maybe not. This [page](http://www.nitrc.org/projects/cluster_roi/) by the authors has Craddock **2011**: 

craddock_2011

File: nilearn/datasets.py [hunk 6]
```diff
@@ -529,6 +529,38 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
...
+        fname = 'smith'
+    elif ds_name == 'haxby2001_simple' or ds_name == 'haxby2001':
+        fname = 'haxby'
+    elif ds_name == 'nyu_rest':
+        fname = 'nyu'
+    elif ds_name == 'miyawaki2008':
+        fname = 'miyawaki'
+    else:
+        descr = ''
+
```
Comment (lesteve) [inline nilearn/datasets.py]: I find these things easier to read with a dict:

```python

dataset_to_filename = {
    'yeo_2011': 'yeo',
    'craddock_2012': 'craddock',
    .
    .
    .
    }

fname = dataset_to_filename.get(ds_name, '')
```

File: nilearn/datasets.py [hunk 7]
```diff
@@ -529,6 +529,38 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
...
+    try:
+        with open(os.path.join(module_path, 'description', fname + '.rst'))\
+                as rst_file:
+            descr = rst_file.read()
+    except Exception as e:
```
Comment (lesteve) [inline nilearn/datasets.py]: You probably want to use something like OSError here.

File: nilearn/datasets.py (hunk 6 above)
Comment (AlexandreAbraham) [inline nilearn/datasets.py]: This should not exist. The name of the files should be the name of the dataset.

File: nilearn/datasets.py (hunk 4 above)
Comment (AlexandreAbraham) [inline nilearn/datasets.py]: I used the year specified on NITRC but @banilo is right when he says that the paper has been published in 2012. Let's ask @GaelVaroquaux to take the final decision ;)

Comment (banilo) [inline nilearn/datasets.py]: > @AlexandreAbraham for consistency reasons, should we change 2011 to 2012 in nitrc too? I guess we'd want to keep the old (i.e. 2011) naming for a while for backward compatibility reasons ...

Does not matter - that is just some website. We should use the date when the paper in qeustion has been actually published in the physical world - and that is 2012. The delay is probably explained by the difference between paper accept and publish date.

File: nilearn/datasets.py (hunk 5 above)
Comment (AlexandreAbraham) [inline nilearn/datasets.py]: I'm not a big fan of exclamation mark in error messages. Is it necessary?

File: nilearn/datasets.py (hunk 6 above)
Comment (banilo) [inline nilearn/datasets.py]: Good point.
:octocat:

**Comment by banilo** [comment]
:octocat: says "Merge this one!"

**Comment by banilo** [comment]
:camel: says "Octocat is right."

**Comment by banilo** [comment]
:neckbeard: : "totally!"

**Comment by banilo** [comment]
:family: : "We could not agree more."

**Comment by AlexandreAbraham** [comment]
:wave::cop: Tsssk... No!

**Comment by banilo** [comment]
:rabbit: "Huh, whhaaattteee?"

**Pull request merged**